### PR TITLE
Metadata tests followup

### PR DESCRIPTION
**Pull request by ivanayov**
This change completes the metadata api test coverage with reference to the python-tuf implementation, excluding few use-cases that are not applicable

Provides two small metadata fixes and an improved test repository data.

### CODE REVIEW

File: metadata/metadata.go [hunk 1]
```diff
@@ -583,6 +584,11 @@ func (role *SuccinctRoles) GetRoles() []string {
 	return res
 }
 
+func (role *SuccinctRoles) GetSuffixLen() int {
```
Comment (rdimitrov) [inline metadata/metadata.go:587]: Why do we add this?

File: metadata/metadata.go [hunk 2]
```diff
@@ -674,9 +680,11 @@ func (signed *TargetsType) AddKey(key *Key, role string) error {
 	// standard delegated roles
 	if signed.Delegations.Roles != nil {
 		// loop through all delegated roles
+		isDelegatedRole := false
```
Comment (rdimitrov) [inline metadata/metadata.go:683]: Why is this change needed?

File: metadata/metadata_test.go [hunk 3]
```diff
@@ -29,13 +29,13 @@ const TEST_REPOSITORY_DATA = "../testutils/repository_data/repository/metadata"
 
 var fixedExpire = time.Date(2030, 8, 15, 14, 30, 45, 100, time.UTC)
 
-func getSignatureByKeyID(signatures []Signature, keyID string) HexBytes {
-	for _, sig := range signatures {
+func getSignatureByKeyID(signatures []Signature, keyID string) (HexBytes, int) {
```
Comment (rdimitrov) [inline metadata/metadata_test.go:32]: Why do you need the slice index?

**Review by rdimitrov — COMMENTED** [review]
Thanks for the follow-up. Can you also check and address why the CI fails?

File: metadata/metadata_test.go (hunk 3 above)
Comment (ivanayov) [reply metadata/metadata_test.go:32]: It's for [TestMetadataVerifyDelegate](https://github.com/rdimitrov/go-tuf-metadata/pull/54/files/97cd3616a9a36bc4f60d7a7b64d07aba11f95ba2#diff-febffd881a5f8de233564e404dab90eeb9e81e428dbba2c6e4c8665be7aba6acR496) . This allows verifying Signature per key ID.

File: metadata/metadata.go (hunk 2 above)
Comment (rdimitrov) [reply metadata/metadata.go:683]: Nice! Thanks for catching this 💯 👍

File: metadata/metadata_test.go (hunk 3 above)
Comment (rdimitrov) [reply metadata/metadata_test.go:32]: Got it. I see it's mainly used for understanding if it failed so can we avoid using the index of -1 for that and assert if the byte slice is empty instead?

**Review by rdimitrov — COMMENTED** [review]
Thanks for the feedback 💯 We should be okay to merge it once we resolve the last comment 👍

**Comment by ivanayov** [comment]
Thanks @rdimitrov, just pushed the updates and rebased with the fixed CI.

**Review by rdimitrov — APPROVED** [review]
